pkg/input/container/docker.go

```diff
@@ -170,50 +174,63 @@ func (dt *DockerTailer) forwardMessages() {
 			return
 		}
 
-		ts, updatedMsg := dt.updatedDockerMessage(msg.Content())
+		ts, sev, updatedMsg := dt.parseMessage(msg.Content())
 
 		containerMsg := message.NewContainerMessage(updatedMsg)
 		msgOrigin := message.NewOrigin()
 		msgOrigin.LogSource = dt.source
 		msgOrigin.Timestamp = ts
 		msgOrigin.Identifier = dt.Identifier()
+		containerMsg.SetSeverity(sev)
+		containerMsg.SetTagsPayload(dt.tagsPayload)
 		containerMsg.SetOrigin(msgOrigin)
 		dt.outputChan <- containerMsg
 	}
 }
 
-func (dt *DockerTailer) updatedDockerMessage(msg []byte) (string, []byte) {
-	tags, err := tagger.Tag(dockerutil.ContainerIDToEntityName(dt.containerName), false)
+func (dt *DockerTailer) keepDockerTagsUpdated() {
+	dt.checkForNewDockerTags()
+	ticker := time.NewTicker(tagsUpdatePeriod)
+	for _ = range ticker.C {
+		if dt.shouldStop {
+			return
+		}
+		dt.checkForNewDockerTags()
+	}
+}
+
+func (dt *DockerTailer) checkForNewDockerTags() {
+	tags, err := tagger.Tag(dockerutil.ContainerIDToEntityName(dt.containerName), true)
 	if err != nil {
 		log.Println(err)
+	} else {
+		if !reflect.DeepEqual(tags, dt.containerTags) {
+			dt.containerTags = tags
```

Comment:
Should those two fields be updated atomically or it's ok?

Reply:
we never really use `containerTags` elsewhere than for this comparison that happens every 10 seconds, so this one should be fine.
For `tagsPayload`, I was wondering. I think it's ok if one or two additional messages are sent while this is being updated, as it's not real time anyway.